crypto/ec/build.info

```diff
@@ -31,6 +31,13 @@ IF[{- !$disabled{asm} -}]
   $ECDEF_armv4=ECP_NISTZ256_ASM
   $ECASM_aarch64=ecp_nistz256.c ecp_nistz256-armv8.S
   $ECDEF_aarch64=ECP_NISTZ256_ASM
+  IF[{- !$disabled{'sm2'} -}]
+    $ECASM_aarch64=$ECASM_aarch64 ecp_sm2p256.c ecp_sm2p256-armv8.S
```

Comment:
Should ecp_sm2p256.c and  ecp_sm2p256-armv8.S be also conditional on `!$disabled{'sm2-precomp'}`? I assume these are unusable without the table.

I.e., there should be only a single `IF [{- !$disabled{'sm2'} && !$disabled{'sm2-precomp'} -}]` shouldn't it?

Reply:
In fact, SM2 algorithm is also available if there is no precomputed table. As shown in the following code, when there is no precomputed table, the program will use `ecp_sm2p256_windowed_mul()` to calculate the scalar multiplication of the base point.

```
#if !defined(OPENSSL_NO_SM2_PRECOMP)
        if (ecp_sm2p256_is_affine_G(generator)) {
            ecp_sm2p256_point_G_mul_by_scalar(&p.p, k);
        } else
#endif
        {
            /* if no precomputed table */
            const EC_POINT *new_generator[1];
            const BIGNUM *g_scalars[1];

            new_generator[0] = generator;
            g_scalars[0] = scalar;

            if (!ecp_sm2p256_windowed_mul(group, &p.p, g_scalars, new_generator,
                                          (new_generator[0] != NULL
                                           && g_scalars[0] != NULL), ctx))
                goto err;
        }
```

Reply:
I've updated the performance data for the no pre-computed table in the first comment.